lib/settings.coffee

```diff
@@ -15,9 +15,34 @@ Settings =
       unless chrome.runtime.lastError
         @handleUpdateFromChromeStorage key, value for own key, value of items
 
+        @loadSettingsFromLocalStorage()
+
       chrome.storage.onChanged.addListener (changes, area) =>
         @propagateChangesFromChromeStorage changes if area == "sync"
 
+  loadSettingsFromLocalStorage: ->
+    chrome.storage.local.get null, (items) =>
+      unless chrome.runtime.lastError
+        overriddenBySync = []
+        for own key, value of items
+          if @shouldSyncKey key
+            if key of @cache
+              overriddenBySync.push key # This key has already been set from chrome.storage.sync.
+            else
+              @handleUpdateFromChromeStorage key, value
+
+        # All of the values in overriddenBySync have equivalents in chrome.storage.sync which take
+        # priority, so we remove them.
+        chrome.storage.local.remove overriddenBySync if Utils.isBackgroundPage()
+
+      chrome.storage.onChanged.addListener (changes, area) =>
+        return unless area == "local"
+        for key, change of changes
```

Comment:
> I'll tweak the code to deal with your other concerns here.

I'm not sure we should be doing this at all, @mrmr1993.  This is introducing too much additional complexity for very little benefit.  We should probably just force an update in synced storage for the offending settings and be done with it.